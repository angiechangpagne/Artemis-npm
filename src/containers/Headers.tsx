import * as React from 'react';
import Dropdown from '../components/DropdownMenu';

export interface iHeaders {
  visualizerToggle: () => void,
  schemaToggle: () => void,
  cacheToggle: () => void,
}

const Headers: React.FC<iHeaders> = ({ schemaToggle, cacheToggle, visualizerToggle }) => (
  <React.Fragment>
    <div className="hOne">
      <h1>ARTEMIS</h1>
    </div>
    <div className="header-container">
      <Dropdown />
      <button className="header-item item4" onClick={() => visualizerToggle()}>
        Visualizer
      </button>
      <button className="header-item item2" onClick={() => schemaToggle()}>
        Schema 
      </button>
      <button className="header-item item3" onClick={() => cacheToggle()}>
        Cache 
      </button>
    </div>
  </React.Fragment>
)

export default Headers;