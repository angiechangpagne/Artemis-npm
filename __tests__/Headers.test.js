import React from 'react';
import { shallow } from "enzyme";
import Headers from '../src/containers/Headers';
import Dropdown from '../src/components/DropdownMenu';

const Enzyme = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new Adapter() });

const func = jest.fn();

describe('Headers />', () => {
  let props;
  beforeAll(() => {
    props = {
      visualizerToggle: func,
      schemaToggle: func, 
      cacheToggle: func
    }
  })
  it("Should make a snapshot", () => {
    const wrapper = shallow(<Headers {...props} />)
    expect(wrapper).toMatchSnapshot();
  })
  it("Should have <h1>ARTEMIS</h1>", () => {
    const wrapper = shallow(<Headers {...props} />)
    expect(wrapper.ontainers(<h1>ARTEMIS</h1>)).toEqual(true)
  })
  it("Should have two div tag", () => {
    const wrapper = shallow(<Headers {...props} />)
    expect(wrapper.find('div').length).toEqual(2)
  })
  it("Should render <Dropdown />", () => {
    const wrapper = shallow(<Headers {...props} />)
    expect(wrapper.find('button').length).toEqual(3)
  })
  it("Should render three button tag", () => {
    const wrapper = shallow(<Headers {...props} />)
    expect(wrapper.find('button').length).toEqual(3)
  })
})