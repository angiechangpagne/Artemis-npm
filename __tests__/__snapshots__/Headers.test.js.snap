//Jest Snapshot

exports[`<Headers /> Should make a snapshot 1`] = `
<Fragment>
  <div
    className="hOne"
    >
    <h1> 
      ARTEMIS
    </h1>
  </div>
  <div
    className="header-container"
    >

    <_default />
    <button 
      className="header-